keyboards/nullbitsco/snap/snap.c

```diff
@@ -0,0 +1,142 @@
+    matrix_init_remote_kb();
+    matrix_init_user();
+}
+
+void keyboard_post_init_kb(void) {
+    #ifdef CONSOLE_ENABLE
+    debug_enable = true;
+    debug_matrix = true;
+    #endif
+    keyboard_post_init_user();
+}
+
+void matrix_scan_kb(void) {
+    matrix_scan_remote_kb();
+    matrix_scan_user();
+
+    if (is_alt_tab_active) {
+        if (timer_elapsed(alt_tab_timer) > 1000) {
+            unregister_code(KC_LALT);
+            is_alt_tab_active = false;
+        }
+    }
+}
+
+// Use Bit-C LED to show CAPS LOCK and NUM LOCK status
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if (res) {
+        if (is_keyboard_master()) set_bitc_LED(led_state.caps_lock ? LED_DIM : LED_OFF);
+    }
+    return res;
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    // If console is enabled, it will print the matrix position and status of each key pressed
+    #ifdef CONSOLE_ENABLE
+    dprintf("kc: 0x%04X, col: %u, row: %u, pressed: %b, time: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed, record->event.time);
+    #endif
+
+    process_record_remote_kb(keycode, record);
+    if (!process_record_user(keycode, record)) return false;
+
+    switch (keycode) {
+        case RESET:
+            if (record->event.pressed && is_keyboard_master()) {
+                set_bitc_LED(LED_DIM);
+                #ifdef RGBLIGHT_ENABLE
+                rgblight_disable_noeeprom();
+                #endif
+                #ifdef OLED_ENABLE
+                oled_off();
+                #endif
+                bootloader_jump();  // jump to bootloader
+            }
+            return false;
+
+        case DISC_MUTE:
+            if (record->event.pressed) {
+                tap_code(KC_F23);
+                #ifdef RGBLIGHT_ENABLE
+                if (!rgblight_is_enabled()) break;
+
+                if (muted) {
+                    rgblight_enable_noeeprom();
+                } else {
+                    rgblight_timer_disable();
+                    uint8_t val = rgblight_get_val();
+                    rgblight_sethsv_range(255, 255, val, 1, 5);
+                }
+                #endif
+                muted = !muted;
+            }
+            break;
+
+        case SUPER_ALT_TAB:
+            if (record->event.pressed) {
+                if (!is_alt_tab_active) {
+                    is_alt_tab_active = true;
+                    register_code(KC_LALT);
+                }
+                alt_tab_timer = timer_read();
+                register_code(KC_TAB);
+            } else {
+                unregister_code(KC_TAB);
+            }
+            break;
+
+        default:
+            break;
+    }
+
+    return true;
+}
+
+// Workaround for sleep issues
+void suspend_power_down_kb(void) {
+    suspend_power_down_user();
+
+    // Turn off underglow
+#    if defined(RGBLIGHT_SLEEP) && defined(RGBLIGHT_ENABLE)
+    rgblight_suspend();
+#    endif
```

Comment:
This is handled by the core, and shouldn't be needed here. 
```suggestion
```

Reply:
Originally I added the powerdown/suspend code here + NO_SUSPEND_POWER_DOWN because of a bug in AVR sleep -- after some period of time in sleep the boards would get into a strange state that required a full power cycle to get back out of, and in a few cases somehow corrupting the firmware and requiring a reflash from toolbox. After skipping the avr sleep, the issues stopped.

It looks like https://github.com/qmk/qmk_firmware/pull/14210 pulled the sleep code out from the platform-specific code and https://github.com/qmk/qmk_firmware/pull/13918 moved the avr suspend code to platforms/avr, but it's still enabled there.

I have the same code across our other open PRs as well, so I'll wait and see what the resolution is here before fixing there.